spec/use_cases/reservations/create_at_desk_spec.rb

```diff
@@ -0,0 +1,56 @@
+require 'rails_helper'
+
+RSpec.describe Reservations::UseCases::CreateAtDesk do
+  let(:seats) { [10, 11, 12, 13] }
+  let(:movie) { create(:movie) }
+  let(:hall) { create(:hall) }
+  let(:screening) { create(:screening, hall_id: hall.id, movie_id: movie.id, start_time: Date.current + 2.days) }
+  let(:params) do
+    {
+      screening_id: screening.id,
+      seats: ,
+      status: 2
+
+    }
+  end
+
+  let(:instance) { described_class.new(**params) }
+
+  describe '.call' do
+    it 'creates reservation' do
+      expect { instance.call }.to change(Reservation, :count).by(1)
+    end
+
+    it 'creates tickets' do
+      expect { instance.call }.to change(Ticket, :count).by(4)
+    end
+
+    it 'returns reservation with tickets' do
+      expect(instance.call.tickets.count).to eq(4)
+    end
+
+    it 'returns reservation with proper attributes' do
+      expect(instance.call).to have_attributes(screening_id: screening.id, status: 'accepted')
+    end
+
+    it 'returns reservation without errors' do
+      expect(instance.call.errors).to match_array([])
+    end
+
+    context 'when seats invalid' do
+      let(:params) do
+        {
+          screening_id: screening.id,
+          seats: [],
+          status: 0
+        }
+      end
```

Comment:
If you just need to change what seats are in this hash you can overwrite the seats param
This will change the params hash defined at the top as it will use the seats definition from the nested context
```suggestion
      let(:seats) { [] }
```